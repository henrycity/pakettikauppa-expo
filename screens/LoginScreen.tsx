import * as WebBrowser from 'expo-web-browser'
import React from 'react'
import { StyleSheet, Button } from 'react-native'

import RegistrationModal from '../components/RegistrationModal'
import { Text, View } from '../components/Themed'
import useLogin from '../hooks/useLogin'

WebBrowser.maybeCompleteAuthSession()

export default function LoginScreen(): JSX.Element {
  const { login, disabled } = useLogin()

  return (
    <View style={styles.container}>
      <Text style={styles.title}>Login or register with a Google account</Text>
      <View
        style={styles.separator}
        lightColor="#eee"
        darkColor="rgba(255,255,255,0.1)"
      />

      <Button
        disabled={disabled}
        accessibilityLabel="Login button"
        title="Login"
        onPress={() => {
          login()
        }}
      />

      <View style={{ height: 10 }} />

      <RegistrationModal />
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  title: {
    fontSize: 20,
    fontWeight: 'bold',
  },
  separator: {
    marginVertical: 30,
    height: 1,
    width: '80%',
  },
})
