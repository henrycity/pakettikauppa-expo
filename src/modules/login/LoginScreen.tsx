import * as WebBrowser from 'expo-web-browser'
import React from 'react'
import { useTranslation } from 'react-i18next'
import { TouchableOpacity } from 'react-native'

import Styles from '../../common/Styles'
import { useThemedColors, View, Text } from '../../common/Themed'
import RegistrationModal from './components/RegistrationModal'
import useLogin from './hooks/useLogin'

WebBrowser.maybeCompleteAuthSession()

export default function LoginScreen(): JSX.Element {
  const { login } = useLogin()
  const { t } = useTranslation()
  const clickHandler = () => {
    login()
  }

  const theme = useThemedColors()
  return (
    <View style={Styles.container}>
      <Text style={Styles.title}>Login or register with a Google account</Text>
      <View
        style={Styles.separator}
        lightColor="#eee"
        darkColor="rgba(255,255,255,0.1)"
      />
      <TouchableOpacity
        onPress={clickHandler}
        style={[Styles.normalButton, { backgroundColor: theme.buttonColor }]}
        accessibilityLabel="Login button"
      >
        <Text style={[Styles.buttonLabel, { color: theme.buttonTextColor }]}>
          {t('LOGIN')}
        </Text>
      </TouchableOpacity>
      <View style={Styles.gap} />

      <RegistrationModal />
    </View>
  )
}
