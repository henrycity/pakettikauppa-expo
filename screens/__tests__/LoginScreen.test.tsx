import React from 'react'
import renderer from 'react-test-renderer'

import LoginScreen from '../LoginScreen'
it('LoginScreen renders correctly', () => {
  const tree = renderer.create(<LoginScreen />).toJSON()

  expect(tree).toMatchSnapshot()
})
