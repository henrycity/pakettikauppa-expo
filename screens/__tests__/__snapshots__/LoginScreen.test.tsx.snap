// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginScreen renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
      },
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#000",
        },
        Object {
          "fontSize": 20,
          "fontWeight": "bold",
        },
      ]
    }
  >
    Login or register with a Google account
  </Text>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#eee",
        },
        Object {
          "height": 1,
          "marginVertical": 30,
          "width": "80%",
        },
      ]
    }
  />
  <View
    accessibilityLabel="Login button"
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        disabled={false}
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Login
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
        },
        Object {
          "height": 10,
        },
      ]
    }
  />
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Register
      </Text>
    </View>
  </View>
  <Modal
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    visible={false}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 20,
            "fontWeight": "bold",
          }
        }
      >
        Please enter your details: 
      </Text>
      <View
        style={
          Object {
            "height": 10,
          }
        }
      />
      <View />
      <Text>
         Email: 
      </Text>
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        defaultValue=""
        onChangeText={[Function]}
        placeholder="Email"
        rejectResponderTermination={true}
        style={
          Object {
            "borderBottomWidth": 1,
            "width": 200,
          }
        }
        underlineColorAndroid="transparent"
      />
      <View
        style={
          Object {
            "height": 10,
          }
        }
      />
      <Text>
         VAT: 
      </Text>
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        defaultValue=""
        onChangeText={[Function]}
        placeholder="VAT"
        rejectResponderTermination={true}
        style={
          Object {
            "borderBottomWidth": 1,
            "width": 200,
          }
        }
        underlineColorAndroid="transparent"
      />
      <View
        style={
          Object {
            "height": 10,
          }
        }
      />
      <View
        accessibilityRole="button"
        accessibilityState={Object {}}
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Submit
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "height": 10,
          }
        }
      />
      <View
        accessibilityRole="button"
        accessibilityState={Object {}}
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Go back
          </Text>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
